### PR TITLE
Feature/convert and load

### LINKED ISSUE #54: Convert to TFRecords and load a dataset
**Is your feature request related to a problem? Please describe.**
For convenience, it would be great to have a function that converts data into TFRecords and returns a TF Dataset.

**Describe the solution you'd like**
`tfrecorder.convert_and_load(...)`

**Describe alternatives you've considered**
No change: When working in a Python environment, user would have to invoke convert and load operations separately.

### PR DESCRIPTION
**Pull request by cfezequiel**
## Description

Implements `convert_and_load` function and updates other high-level APIs:
`create_tfrecords` -> `convert`
`check_tfrecords` -> `inspect`

Fixes #54 

## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix/feature that may cause existing functionality to not work as expected)
- [X] This change requires a documentation update

## Checklist

Please delete options that are not relevant.

- [X] My code adheres to the [Google Python Style Guide](https://google.github.io/styleguide/pyguide.html)
- [X] I ran `make pylint` and code is rated 10/10
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I ran `make test` and `make testnb` and all tests pass
- [X] I ran the tool and verified the change works
- [X] I have adequately commented my code, particularly in hard-to-understand areas
- [X] I have made relevant changes to the documentation, if needed
- [X] My changes generate no new warnings
- [X] I have corrected any misspellings in my code
- [X] (For feature) I made sure that I'm merging to the `dev` branch

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -23,7 +23,7 @@ Please delete options that are not relevant.
 - [ ] My code adheres to the [Google Python Style Guide](https://google.github.io/styleguide/pyguide.html)
 - [ ] I ran `make pylint` and code is rated 10/10
 - [ ] I have added tests that prove my fix is effective or that my feature works
-- [ ] I ran `make test` and all tests pass
+- [ ] I ran `make test testnb` and all tests pass
```
Comment (mbernico) [inline .github/PULL_REQUEST_TEMPLATE.md:26]: Should we do something like make test-all or even have make test run all tests and then use another make command to test only .py files?

File: README.md [hunk 2]
```diff
@@ -187,8 +187,9 @@ representing the images encoded into TFRecords.
 Using the command line:
 
 ```bash
-tfrecorder check-tfrecords \
-    --file_pattern=/path/to/tfrecords/train*.tfrecord.gz \
+tfrecorder inspect \
+    --tfrecord-dir=/path/to/tfrecords/ \
+    --split='TRAIN' \
```
Comment (mbernico) [inline README.md:192]: I like this much better!

File: tfrecorder/utils_test.py [hunk 3]
```diff
@@ -27,10 +27,11 @@
 import tensorflow as tf
 
 from tfrecorder import beam_image
-from tfrecorder import check
+from tfrecorder import constants
+from tfrecorder import utils
 from tfrecorder import test_utils
 from tfrecorder import input_schema
-from tfrecorder import dataset as _dataset
+from tfrecorder import dataset_loader as _dataset
```
Comment (mbernico) [inline tfrecorder/utils_test.py:34]: Is _dataset still required here?

**Review by mbernico — COMMENTED** [review]
Nice job, I really like these changes.  One minor idea, one nit.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (cfezequiel) [reply .github/PULL_REQUEST_TEMPLATE.md:26]: As per chat, added `test-nb` and `test-py` for notebook and py file tests, and changed `test` to run them all.

File: README.md (hunk 2 above)
Comment (cfezequiel) [reply README.md:192]: Cool! I was also thinking we could maybe change the default behavior to dump some data for each split in a future PR.

File: tfrecorder/utils_test.py (hunk 3 above)
Comment (cfezequiel) [reply tfrecorder/utils_test.py:34]: Good catch. Eliminated the alias.

**Review by mbernico — APPROVED** [review]

**Pull request merged**